packages/account/src/providers/resource-cache.ts

```diff
@@ -26,54 +32,99 @@ export class ResourceCache {
   }
 
   // Add resources to the cache
-  set(transactionId: string, resources: Required<ExcludeResourcesOption>): void {
-    const currentTime = Date.now();
-    const existingResources = cache.get(transactionId) || {
-      utxos: new Set<string>(),
-      messages: new Set<string>(),
-      timestamp: currentTime,
-    };
-
-    resources.utxos.forEach((utxo) => existingResources.utxos.add(hexlify(utxo)));
-    resources.messages.forEach((message) => existingResources.messages.add(hexlify(message)));
-
-    cache.set(transactionId, existingResources);
+  set(transactionId: string, inputs: TransactionRequestInput[]): void {
+    const transactionResourceCache = this.setupResourcesCache(inputs);
+    cache.set(transactionId, transactionResourceCache);
   }
 
-  // Remove resources from the cache for a given transaction ID
   unset(transactionId: string): void {
     cache.delete(transactionId);
   }
 
-  // Get all cached resources and remove expired ones
-  getActiveData() {
-    const allResources: { utxos: string[]; messages: string[] } = { utxos: [], messages: [] };
+  getActiveData(owner: string) {
+    const activeData: { utxos: string[]; messages: string[] } = { utxos: [], messages: [] };
     const currentTime = Date.now();
+    const expired: string[] = [];
+
     cache.forEach((resource, transactionId) => {
-      if (currentTime - resource.timestamp < this.ttl) {
-        allResources.utxos.push(...resource.utxos);
-        allResources.messages.push(...resource.messages);
+      const isActive = currentTime - resource.timestamp < this.ttl;
+
+      if (isActive) {
+        const resourcesFromOwner = resource.owners.get(owner);
+        if (resourcesFromOwner) {
+          activeData.utxos.push(...resourcesFromOwner.utxos);
+          activeData.messages.push(...resourcesFromOwner.messages);
+        }
       } else {
-        cache.delete(transactionId);
+        expired.push(transactionId);
       }
     });
-    return allResources;
+
+    expired.forEach(this.unset);
+
+    activeData.utxos.reverse();
+    activeData.messages.reverse();
+
+    return activeData;
   }
 
-  // Check if a UTXO ID or message nonce is already cached and not expired
-  isCached(key: string): boolean {
+  isCached(owner: string, key: string): boolean {
```

Comment:
In the PR description, I thought this method accepted an object.

Perhaps removing the `:string` type notation can clear up this impression.


```ts
// before
provider.cache?.getActiveData();
provider.cache?.isCached(key);
```


```ts
// after
const owner = wallet.address.toB256();

provider.cache?.getActiveData(owner);
provider.cache?.isCached(key, owner);
```